src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,203 @@
+                    + order.getUserId(), ex);
+        }
+        insertProductsToOrder(order);
+        return order;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ? AND deleted = FALSE";
+        Order order = null;
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = getOrderFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get order with id = "
+                    + id, ex);
+        }
+
+        if (order != null) {
+            order.setProducts(getListOfProducts(order.getId()));
+        }
+        return Optional.ofNullable(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT * FROM orders WHERE deleted = FALSE";
+        List<Order> orders = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrderFromResultSet(resultSet);
+                orders.add(order);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get list of all orders", ex);
+        }
+
+        for (Order order: orders) {
+            order.setProducts(getListOfProducts(order.getId()));
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "UPDATE orders SET user_id = ? "
+                + "WHERE order_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getUserId());
+            statement.setLong(2, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update order with id = "
+                    + order.getId(), ex);
+        }
+
+        removeAllProductsFromOrder(order.getId());
+        insertProductsToOrder(order);
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders SET deleted = TRUE WHERE order_id = ?";
+        int itemsDeleted = 0;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            itemsDeleted = statement.executeUpdate();
```

Comment:
You can do - return statement.executeUpdate() == 1; and then you don't need to declare itemsDeleted...